src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1265,106 +1280,133 @@ protected override void ProcessRecord()
                             WriteVerbose(linkVerboseMsg);
                         }
 
-                        using (HttpRequestMessage request = GetRequest(uri, stripAuthorization:false))
+                        int retriesLeft = RetryCount;
+                        HttpResponseMessage response = null;
+
+                        bool isSuccessful = true;
+
+                        do
```

Comment:
Can you investigate collapsing this to a single do/while instead of nested?

Reply:
Moved logic to GetResponse()